standalone/portal-api/src/modules/company/services/supplier-entity.service.ts

```diff
@@ -0,0 +1,383 @@
+import { SupplierCompanyInformationRepository } from 'src/modules/company-information/repositories/supplier.company.information.repository';
+import { UpdateSupplierEntityDto } from '../dto/request/update-supplier-entity.dto';
+import { CompanyTaxAndRevenueRepository } from 'src/modules/company-information/repositories/company.information.tax.and.revenue.repository';
+
+@Injectable()
+export class SupplierCompanyEntityService {
+  constructor(
+    @InjectRepository(SupplierCompanyEntity)
+    private supplierCompanyRepository: SupplierCompanyEntityRepository,
+
+    @InjectRepository(CompanyAddressEntity)
+    private companyAddressRepository: CompanyAddressRepository,
+
+    @InjectRepository(CompanyTaxAndRevenueEntity)
+    private companyTaxAndRevenueRepository: CompanyTaxAndRevenueRepository,
+
+    @InjectRepository(SupplierCompanyInformationEntity)
+    private supplierCompanyInformationRepository: SupplierCompanyInformationRepository,
+
+    private mappingService: EntityMappingService,
+  ) {}
+
+  public async findSupplierEntityById(
+    id: string,
+  ): Promise<SupplierCompanyEntity> {
+    if (!id) throw triggerError('missing-entity-id');
+
+    const entity = await this.supplierCompanyRepository.findOne(id, {
+      relations: [
+        'relationshipWithBuyers',
+        'applications',
+        'companyInformation',
+        'companyInformation.address',
+        'companyInformation.taxAndRevenue',
+      ],
+    });
+    console.log('entity', entity);
+    if (!entity) throw triggerError('entity-not-found');
+
+    return entity;
+  }
+
+  public async deleteSupplierEntity(id: string): Promise<void> {
+    if (!id) throw triggerError('missing-entity-id');
+
+    const entity = await this.supplierCompanyRepository.findOneOrFail({
+      where: { id: id },
+    });
+
+    if (!entity) throw triggerError('entity-not-found');
+
+    await this.supplierCompanyRepository.delete({ id: id });
+  }
+
+  public async getSupplierEntity(
+    id: string,
+  ): Promise<GetSupplierEntityDetailsResponse> {
+    if (!id) throw triggerError('missing-entity-id');
+
+    const entity = await this.supplierCompanyRepository.findOne({
+      relations: [
+        'relationshipWithBuyers',
+        'companyInformation',
+        'applications',
+        'companyInformation.address',
+        'companyInformation.taxAndRevenue',
+      ],
+      where: { id: id },
+    });
+
+    if (!entity) throw triggerError('entity-not-found');
+
+    return this.mappingService.mapSupplierEntityDetails(entity);
+  }
+  public async getAllSupplierEntity(): Promise<SupplierCompanyEntity[]> {
+    return await this.supplierCompanyRepository.find({
+      select: ['id', 'name', 'abbreviation', 'logo'],
+    });
+  }
+
+  @Transactional()
+  public async updateSupplierEntity(
+    id: string,
+    data: UpdateSupplierEntityDto,
+  ): Promise<GetSupplierEntityDetailsResponse> {
+    runOnTransactionRollback((cb) =>
```

Comment:
For both transactional hooks, please replace with the helper method on the BaseService